sdks/python/apache_beam/runners/dataflow/internal/dependency.py

```diff
@@ -547,8 +547,12 @@ def _get_required_container_version(job_type=None):
     version = pkg_resources.get_distribution(GOOGLE_PACKAGE_NAME).version
     # We drop any pre/post parts of the version and we keep only the X.Y.Z
     # format.  For instance the 0.3.0rc2 SDK version translates into 0.3.0.
-    container_version = (
-        '%s.%s.%s' % pkg_resources.parse_version(version)._version.release)
+    parsed_version = pkg_resources.parse_version(version)
```

Comment:
Could you add a comment here about why are we doing this, and for what version of the pkg_resources?

Reply:
I'll check when I'm back

Reply:
FWIW,
test_harness_override_present_in_dataflow_distributions passes for me on master for Python 2.7.6 and setuptools==36.7.2. 

I would prefer to use 
`container_version=pkg_resources.get_distribution(GOOGLE_PACKAGE_NAME).version` and not parse it further. We had a conversation earlier with @aaltay that there is currently no need to for additional parsing; we don't release "-RC" versions of the SDK, and it is reasonable to require 1-1 correspondence between container name and SDK version if we did.

Reply:
So it appears to be an old version of setuptools, however its still the default one in ubuntu LTS. We could also fix this by requiring a new setuptools in setup.py (either way)

Reply:
I think we should prefer using specifying a >= version for setuptools, if possible, instead of continuing to maintain workarounds like these.